app/models/course/assessment.rb

```diff
@@ -127,6 +127,14 @@ def password_protected?
     password.present?
   end
 
+  def submission_downloadable?
+    questions.where(
```

Comment:
`downloadable?` is implemented in answer model and used when downloading the individual answers. Here the answers are not accessed at all, the question types is used to check if submissions can be downloaded.